jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/selection/BuildingSelectionContent.java

```diff
@@ -332,8 +332,15 @@ private BuildingBackgroundPanel createNormalBuildingContent(BuildingState state)
 			layout.priority.setPriority(supported, building.getPriority());
 		}
 
-		if (building.getBuildingType().getWorkRadius() <= 0) {
-			layout.background.removeChild(layout.workRadius);
+		if (building.getBuildingType() == EBuildingType.DOCKYARD) {
+			layout.background.removeChild(layout.buttonWorkRadius);
+		} else {
+			layout.background.removeChild(layout.buttonDockPosition);
```

Comment:
I prefer to make a secial layout for special buildings. The problem is, that we need some other buttons for the catapult factory, ...
You can simply create a new XML layout file and load it. See the trading buildings for examples.